### PR TITLE
add opt-in stream store block compression

### PR DESCRIPTION
**Pull request by clockfort**
(internal ticket QA-91866)

Contains an implicit assumption (via compression block size setting)
that despite using the stream store users will very often have
stored streams of less than a single block in size.

This block size is meant to be an adequate balance between overfetching
beyond a stream block, being able to build up an adequate dictionary
for a good compression ratio, and the expected long tail of many-block
large stream fetches.

We could open this up for more direct user configurability,
but in all cases I could think of except laziness, the user who
would most benefit from messing with the compression block size
should actually not use this feature and instead implement compression
on their value level themselves.

### CODE REVIEW

File: atlasdb-client/src/main/java/com/palantir/atlasdb/schema/stream/StreamTables.java [hunk 1]
```diff
@@ -32,9 +32,10 @@
     public static final String HASH_TABLE_SUFFIX = "_stream_hash_idx";
     public static final String INDEX_TABLE_SUFFIX = "_stream_idx";
 
+    // todo can we remove this or at least rename it? It only generates the schema as if you had accepted all defaults and callers using the returned schema for other things could unknowingly blow away their changes
     public static void generateSchema(Schema schema, final String shortPrefix, final String longPrefix, final ValueType idType) {
         schema.addTableDefinition(shortPrefix + METADATA_TABLE_SUFFIX, getStreamMetadataDefinition(longPrefix, idType, ExpirationStrategy.NEVER, false, false));
-        schema.addTableDefinition(shortPrefix + VALUE_TABLE_SUFFIX, getStreamValueDefinition(longPrefix, idType, ExpirationStrategy.NEVER, false, false));
+        schema.addTableDefinition(shortPrefix + VALUE_TABLE_SUFFIX, getStreamValueDefinition(longPrefix, idType, ExpirationStrategy.NEVER, false, false, false));
```
Comment (schlosna) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/schema/stream/StreamTables.java]: Boolean args are getting a bit unreadable, time for enums?

Comment (clockfort) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/schema/stream/StreamTables.java]: agreed, but probably by Tanya who is making other more major stream store improvements already. Noticed a couple other things that are gross / need cleanup into separate classes.

Comment (clockfort) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/schema/stream/StreamTables.java]: I can do some of them now though, I think she'll forgive me for at least a few merge conflicts.

**Comment by clockfort** [comment]
not super happy with the cleanup commit, can't think of a particularly pleasant way to make most callers actually use the builder itself except by them bypassing the logic in Schema#addStreamStore

Welcome to suggestions

**Comment by clockfort** [comment]
Maybe make a more overall StreamStore builder that internally describes all of these tables and making people just attach the built version of to their Schema with 
Schema.addStreamStore(new StreamStoreBuilder(foo).withBarFeature().build());

**Comment by tmgordeeva** [comment]
yeah definitely +1 to a builder, too many "boolean, boolean, boolean" for this not to get confusing and have bugs eventually

**Comment by clockfort** [comment]
Sure. Do you see an easy way to make the builder pattern extend up into Schema level?
Because right now the builder is at too low a level for regular people to even see that it's making things slightly better :-/

File: atlasdb-client/src/main/java/com/palantir/atlasdb/schema/stream/StreamStoreDefinition.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    public Map<String, TableDefinition> getTables() {
+        return streamStoreTables;
+    }
+
+    public StreamStoreRenderer getRenderer(String packageName, String name) {
+        return new StreamStoreRenderer(Renderers.CamelCase(longName), idType, packageName, name, inMemoryThreshold, expirationStrategy);
+    }
+
+    public String shortName() {
```
Comment (clockfort) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/schema/stream/StreamStoreDefinition.java]: probably going to swap this out to be getCleanupTasks, just getting lazy with how much work the refactor has been

**Comment by clockfort** [comment]
So uh, the error the tests are running into is legitimate. But confusing.
"Stream ids must be a long for persistent streams", is a check that I moved slightly to happen a little before where it used to do the check.  But we have the example schema try and persistent stream w/ VAR_LONG.
Not sure how this didn't fail before?
??
Any ideas @tmgordeeva

**Pull request merged**